akka-stream-typed/src/main/scala/akka/stream/typed/javadsl/ActorFlow.scala

```diff
@@ -134,4 +135,48 @@ object ActorFlow {
       .askWithStatus[I, Q, A](parallelism)(ref)((i, ref) => makeMessage(i, ref))(timeout.toMillis.millis)
       .asJava
 
+  /**
+   * Use the `ask` pattern to send a request-reply message to the target `ref` actor without including the context.
+   */
+  def askWithContext[I, Q, A, Ctx](
+      ref: ActorRef[Q],
+      timeout: java.time.Duration,
+      executionContextExecutor: ExecutionContextExecutor,
+      makeMessage: BiFunction[I, ActorRef[A], Q]): Flow[(I, Ctx), (A, Ctx), NotUsed] =
```

Comment:
Should be `Pair` rather than tuples for the Java APIs

Reply:
Ok, fixed... Not sure if there is a more convenient way to convert from scala `Tuple2` to `Pair` and back. I'm not used to work with the Java API.

Reply:
You could potentially use something like `Pair.apply.tupled` but not sure if it is so much cleaner/worth it.

Reply:
ok then, I'll leave it like this if you do not have any preference about it